tlsconfig/config.go

```diff
@@ -96,6 +96,40 @@ func certPool(caFile string, exclusivePool bool) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+// isValidMinVersion checks that the input value is a valid tls minimum version
+func isValidMinVersion(version uint16) bool {
+	allVersions := []uint16{
+		tls.VersionSSL30,
+		tls.VersionTLS10,
+		tls.VersionTLS11,
+		tls.VersionTLS12,
+	}
+
+	for _, v := range allVersions {
+		if version == v {
+			return true
+		}
+	}
+	return false
+}
+
+// adjustMinVersion sets the MinVersion on `config`, the input configuration.
+// It assumes the current MinVersion on the `config` is the lowest allowed.
+func adjustMinVersion(options Options, config *tls.Config) error {
+	if options.MinVersion > 0 {
+		if !isValidMinVersion(options.MinVersion) {
+			return fmt.Errorf("Invalid minimum TLS version: %x", options.MinVersion)
+		}
+		if options.MinVersion >= config.MinVersion {
+			config.MinVersion = options.MinVersion
+		} else {
+			return fmt.Errorf("Requested minimum TLS version is too low.")
```

Comment:
Also not too important, but it's slightly better to `if options.MinVersion < config.MinVersion { return fmt.Errorf(...) }`, it won't require any `else` statement.